docs/sources/flow/reference/components/otelcol.exporter.loadbalancing.md

```diff
@@ -0,0 +1,256 @@
+provided as an IP address (`1.2.3.4`), an IP address in CIDR notation
+(`1.2.3.4/8`), a domain name (`example.com`), or `*`. A domain name matches
+that domain and all subdomains. A domain name with a leading "."
+(`.example.com`) matches subdomains only. `NO_PROXY` is only read when
+`HTTPS_PROXY` is set.
+
+Because `otelcol.exporter.loadbalancing` uses gRPC, the configured proxy server must be
+able to handle and proxy HTTP/2 traffic.
+
+[Load balancing]: https://github.com/grpc/grpc-go/blob/master/examples/features/load_balancing/README.md#pick_first
+[HTTP CONNECT]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/CONNECT
+
+### tls block
+
+The `tls` block configures TLS settings used for the connection to the gRPC
+server.
+
+{{< docs/shared lookup="flow/reference/components/otelcol-tls-config-block.md" source="agent" >}}
+
+### keepalive block
+
+The `keepalive` block configures keepalive settings for gRPC client
+connections.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`ping_wait` | `duration` | How often to ping the server after no activity. | | no
+`ping_response_timeout` | `duration` | Time to wait before closing inactive connections if the server does not respond to a ping. | | no
+`ping_without_stream` | `boolean` | Send pings even if there is no active stream request. | | no
+
+### queue block
+
+The `queue` block configures an in-memory buffer of batches before data is sent
+to the gRPC server.
+
+{{< docs/shared lookup="flow/reference/components/otelcol-queue-block.md" source="agent" >}}
+
+### retry block
+
+The `retry` block configures how failed requests to the gRPC server are
+retried.
+
+{{< docs/shared lookup="flow/reference/components/otelcol-retry-block.md" source="agent" >}}
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`input` | `otelcol.Consumer` | A value that other components can use to send telemetry data to.
+
+`input` accepts `otelcol.Consumer` data for any telemetry signal (metrics,
+logs, or traces).
+
+## Component health
+
+`otelcol.exporter.loadbalancing` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`otelcol.exporter.loadbalancing` does not expose any component-specific debug
+information.
+
+## Example
+
+This example accepts OTLP traces over gRPC. It then sends them in a load-balanced 
+way to "localhost:55690" or "localhost:55700" according to their trace ID.
+
+```river
+otelcol.receiver.otlp "default" {
+    grpc {}
+    output {
+        traces  = [otelcol.exporter.loadbalancing.default.input]
+    }
+}
+
+otelcol.exporter.loadbalancing "default" {
+    resolver {
+        static {
+            hostnames = ["localhost:55690", "localhost:55700"]
+        }
+    }
+    protocol {
+        otlp {
+            client {}
```

Comment:
It's a bit weird how we have to add all these blocks just so `client {}` gets populated with defaults. IMO the user would not expect that. They would expect not having blocks would simply init them to the default values mentioned in the docs. Instead, they'd be zero-initialised. It feels a bit wrong to have to make the blocks mandatory just so they get their default values, and I might raise an issue about this so we can discuss it separately.

The problem is that if they get set to their default values, what should we set the mandatory values to? I guess the argument for not setting defaults is that mandatory attributes are not set either. But then probably nothing should be set and the whole block should be a nil.